lib/jsonapi/acts_as_resource_controller.rb

```diff
@@ -234,7 +234,7 @@ def render_results(operation_results)
       end
 
       render_options[:location] = content[:data]["links"][:self] if (
-        response_doc.status == :created && content[:data].class != Array
+        response_doc.status == :created && content[:data].class != Array && content[:data]["links"]
```

Comment:
This probably isn't strictly necessary, but it shows that `:location` should not be empty:

```suggestion
        response_doc.status == :created && content[:data].class != Array && content[:data]["links"] && content[:data]["links"][:self]
```

Reply:
This just makes the relationships easier to read in the debugger. The `to_s` isn't called at other times. Yes, the ternary was backwards.